pages/matrix-files/[...id].js

```diff
@@ -0,0 +1,210 @@
+import ObjectPageHeader from "../../components/object-page-header";
+import PagePreamble from "../../components/page-preamble";
+import SeparatedList from "../../components/separated-list";
+// lib
+import buildAttribution from "../../lib/attribution";
+import buildBreadcrumbs from "../../lib/breadcrumbs";
+import {
+  requestDocuments,
+  requestFileSets,
+  requestFiles,
+} from "../../lib/common-requests";
+import errorObjectToProps from "../../lib/errors";
+import FetchRequest from "../../lib/fetch-request";
+import {
+  checkForFileDownloadPath,
+  convertFileDownloadPathToFilePagePath,
+} from "../../lib/files";
+import { isJsonFormat } from "../../lib/query-utils";
+
+export default function MatrixFile({
+  attribution,
+  matrixFile,
+  fileSet = null,
+  documents,
+  derivedFrom,
+  derivedFromFileSets,
+  fileFormatSpecifications,
+  referenceFiles,
+  isJson,
+}) {
+  return (
+    <>
+      <Breadcrumbs />
+      <EditableItem item={matrixFile}>
+        <PagePreamble>
+          <AlternateAccessions
+            alternateAccessions={matrixFile.alternate_accessions}
+          />
+        </PagePreamble>
+        <ObjectPageHeader item={matrixFile} isJsonFormat={isJson}>
+          <FileHeaderDownload file={matrixFile} />
+        </ObjectPageHeader>
+        <JsonDisplay item={matrixFile} isJsonFormat={isJson}>
+          <DataPanel>
+            <DataArea>
+              <FileDataItems
+                item={matrixFile}
+                fileSet={fileSet}
+              ></FileDataItems>
+            </DataArea>
+          </DataPanel>
+          <DataAreaTitle>Matrix Details</DataAreaTitle>
+          <DataPanel>
+            <DataArea>
+              <>
+                <DataItemLabel>First Dimension</DataItemLabel>
+                <DataItemValue>{matrixFile.dimension1}</DataItemValue>
+              </>
+              <>
+                <DataItemLabel>Second Dimension</DataItemLabel>
+                <DataItemValue>{matrixFile.dimension2}</DataItemValue>
+              </>
+              {referenceFiles.length > 0 && (
+                <>
+                  <DataItemLabel>Reference Files</DataItemLabel>
+                  <DataItemValue>
+                    <SeparatedList>
+                      {referenceFiles.map((file) => (
+                        <Link href={file["@id"]} key={file["@id"]}>
+                          {file.accession}
+                        </Link>
+                      ))}
+                    </SeparatedList>
+                  </DataItemValue>
+                </>
+              )}
+            </DataArea>
+          </DataPanel>
+          {derivedFrom.length > 0 && (
+            <>
+              <DataAreaTitle>
+                Files {matrixFile.accession} Derives From
+              </DataAreaTitle>
+              <DerivedFromTable
+                derivedFrom={derivedFrom}
+                derivedFromFileSets={derivedFromFileSets}
+              />
+            </>
+          )}
```

Comment:
And to continue to say way too much about this, this is correct — you need this `<>…</>` because this `derivedFrom` loop starts a new JSX expression, so you can’t return both `<DataAreaTitle>` _and_ `<DerivedFromTable>` — two is too many.